src/Features/VisualBasic/Portable/Structure/VisualBasicStructureHelpers.vb

```diff
@@ -72,16 +73,96 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
         End Sub
 
         Friend Sub CollectCommentsRegions(node As SyntaxNode,
-                                          spans As ArrayBuilder(Of BlockSpan))
+                                          spans As ArrayBuilder(Of BlockSpan),
+                                          isMetadataAsSource As Boolean)
             If node Is Nothing Then
                 Throw New ArgumentNullException(NameOf(node))
             End If
 
-            Dim triviaList = node.GetLeadingTrivia()
-
-            CollectCommentsRegions(triviaList, spans)
+            Dim span As BlockSpan = Nothing
+            If isMetadataAsSource AndAlso TryGetLeadingCollapsibleSpan(node, span) Then
+                spans.Add(span)
+            Else
+                Dim triviaList = node.GetLeadingTrivia()
+                CollectCommentsRegions(triviaList, spans)
+            End If
         End Sub
 
+        Private Function TryGetLeadingCollapsibleSpan(node As SyntaxNode, <[Out]> ByRef span As BlockSpan) As Boolean
```

Comment:
pity to have the duplication.  seems sharable.

Reply:
I'm leaving it like this for two reasons:

1. It was already split before this pull request, so it's not a direct move for the worse
2. The main complexity in this code is in `GetEndToken`, which is not as easy to share